Remotion/Data/DomainObjects/Persistence/Configuration/IStorageObjectFactoryFactoryExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+
+namespace Remotion.Data.DomainObjects.Persistence.Configuration
+{
+  public static class IStorageObjectFactoryFactoryExtensions
+  {
+    /// <summary>
+    /// Resolves or creates an <see cref="IStorageObjectFactory"/> object based on the <see cref="Type"/> of the generic parameter.
+    /// </summary>
+    public static T Create<T> (this IStorageObjectFactoryFactory storageObjectFactoryFactory)
```

Comment:
We usually also add arg checks for extension methods. Even for the `this` parameter as it can be null due to the translation. e.g. `value.DoSomething(1)` is translated to `MyExtensions.DoSomething(value, 1)` so if values is null no null ref exception is thrown and the null values is passed to the extension method